File: data/multilang_comparison/program_cpp_vector/src/func29.cpp
#include "program_cpp_vector.hpp" 
Array* func29(Array_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   unsigned int loop5 = 0;
   unsigned int loopLimit5 = (50)/4 + 1;
   for(; loop5 < loopLimit5; loop5++) {
      if(PATH0 & 1) {
         Array* array8;
         if (pCounter > 0) {
            array8 = vars->data.at(--pCounter);
            array8->refC++;
            DEBUG_COPY(array8->id);
         } else {
            array8 = new Array();
            array8->size = 802;
            array8->refC = 1;
            array8->id = 8;
            array8->data = std::vector<unsigned int>(array8->size, 0);
            DEBUG_NEW(array8->id);
         }
         unsigned int loop6 = 0;
         unsigned int loopLimit6 = (50)/5 + 1;
         for(; loop6 < loopLimit6; loop6++) {
            for (int i = 0; i < array8->size; i++) {
                array8->data.at(i)--;
            }
            for (int i = 0; i < array8->size; i++) {
                if (array8->data.at(i) == 69) {
                    return array8;
                }
            }
         }
         array8->refC--;
         if(array8->refC == 0) {
            array8->data.clear();
            delete array8;
            DEBUG_FREE(array8->id);
         }
      }
      else {
         Array_param params0;
         params0.size = 0;
         std::vector<Array*> data_params0(0);
         params0.data = data_params0;
         Array* array9 = func40(&params0, loopsFactor);
         DEBUG_RETURN(array9->id);
         if(params0.size > 0) {
         	 params0.data.clear();
         }
         array9->refC--;
         if(array9->refC == 0) {
            array9->data.clear();
            delete array9;
            DEBUG_FREE(array9->id);
         }
      }
      Array* array11;
      if (pCounter > 0) {
         array11 = vars->data.at(--pCounter);
         array11->refC++;
         DEBUG_COPY(array11->id);
      } else {
         array11 = new Array();
         array11->size = 784;
         array11->refC = 1;
         array11->id = 11;
         array11->data = std::vector<unsigned int>(array11->size, 0);
         DEBUG_NEW(array11->id);
      }
      Array_param params0;
      params0.size = 1;
      std::vector<Array*> data_params0(1);
      params0.data = data_params0;
      params0.data.at(0) = array11;
      Array* array12 = func41(&params0, loopsFactor);
      DEBUG_RETURN(array12->id);
      if(params0.size > 0) {
      	 params0.data.clear();
      }
      for (int i = 0; i < array11->size; i++) {
          if (array11->data.at(i) == 73) {
              return array11;
          }
      }
      array12->refC--;
      if(array12->refC == 0) {
         array12->data.clear();
         delete array12;
         DEBUG_FREE(array12->id);
      }
      array11->refC--;
      if(array11->refC == 0) {
         array11->data.clear();
         delete array11;
         DEBUG_FREE(array11->id);
      }
   }
   Array* array14;
   if (pCounter > 0) {
      array14 = vars->data.at(--pCounter);
      array14->refC++;
      DEBUG_COPY(array14->id);
   } else {
      array14 = new Array();
      array14->size = 862;
      array14->refC = 1;
      array14->id = 14;
      array14->data = std::vector<unsigned int>(array14->size, 0);
      DEBUG_NEW(array14->id);
   }
   return array14;
}

File: data/multilang_comparison/program_julia/src/func29.jl
include("program_julia_head.jl")
function func29(vars::JArray_param, PATH0::UInt64, loopsFactor::Int)::JArray
   pCounter = length(vars.data)
   loopLimit5 = (50)/4 + 1
   for i in 1:loopLimit5 
      if (PATH0 & 1) != 0
         array8 = nothing
         if pCounter > 1
             array8 = vars.data[pCounter]
             pCounter -= 1
             array8.refC += 1
         else
             array8 = JArray(zeros(Int, 802), 802, 1, 8)
         end
         loopLimit6 = (50)/5 + 1
         for i in 1:loopLimit6 
            for i in 0:(array8.size-1) 
               array8.data[i+1] -= 1
            end
            for i in 1:array8.size
                if array8.data[i] == 69
                        return array8
               end
            end
         end
         array8.refC -= 1
         if array8.refC == 0
             empty!(array8.data)
             array8 = nothing
         end
      
      else 
         params0 = JArray_param{Int}()
         params0.size = 0
         array9 = func40(params0, loopsFactor);
         array9.refC -= 1
         if array9.refC == 0
             empty!(array9.data)
             array9 = nothing
         end
      end
      array11 = nothing
      if pCounter > 1
          array11 = vars.data[pCounter]
          pCounter -= 1
          array11.refC += 1
      else
          array11 = JArray(zeros(Int, 784), 784, 1, 11)
      end
      params0 = JArray_param{Int}()
      params0.size = 1
      push!(params0.data, array11)
      params0.size += 1
      array12 = func41(params0, loopsFactor);
      for i in 1:array11.size
          if array11.data[i] == 73
                  return array11
         end
      end
      array12.refC -= 1
      if array12.refC == 0
          empty!(array12.data)
          array12 = nothing
      end
      array11.refC -= 1
      if array11.refC == 0
          empty!(array11.data)
          array11 = nothing
      end
   end
   array14 = nothing
   if pCounter > 1
       array14 = vars.data[pCounter]
       pCounter -= 1
       array14.refC += 1
   else
       array14 = JArray(zeros(Int, 862), 862, 1, 14)
   end
   return array14;
end

